From b48d1c08a007c1683743ef5cf7d6002c365df832 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alejandro=20Gonz=C3=A1lez?=
 <alejandro.gonzalez.correo@gmail.com>
Date: Sat, 8 Aug 2020 15:04:26 +0200
Subject: [PATCH 1/1] package/stress-ng: bump version to 0.11.17
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This commit bumps stress-ng version to 0.11.17.

This version includes the patch that fixes musl build at commit
03416938871388243d28621f4b59ce532231f11c, and also fixes a
"Invalid syntax in conditional" build error in a Makefile.

Signed-off-by: Alejandro González <alejandro.gonzalez.correo@gmail.com>
---
 ...tress-ng.h-suppress-kernel-sysinfo.h.patch | 49 -------------------
 package/stress-ng/stress-ng.hash              |  2 +-
 package/stress-ng/stress-ng.mk                |  2 +-
 3 files changed, 2 insertions(+), 51 deletions(-)
 delete mode 100644 package/stress-ng/0001-stress-ng.h-suppress-kernel-sysinfo.h.patch

diff --git a/package/stress-ng/0001-stress-ng.h-suppress-kernel-sysinfo.h.patch b/package/stress-ng/0001-stress-ng.h-suppress-kernel-sysinfo.h.patch
deleted file mode 100644
index cc8b1a1b67..0000000000
--- a/package/stress-ng/0001-stress-ng.h-suppress-kernel-sysinfo.h.patch
+++ /dev/null
@@ -1,49 +0,0 @@
-From 8c223daf10319f15570a74ed83a1d655fdbd4a7b Mon Sep 17 00:00:00 2001
-Message-Id: <8c223daf10319f15570a74ed83a1d655fdbd4a7b.1592540443.git.baruch@tkos.co.il>
-From: Baruch Siach <baruch@tkos.co.il>
-Date: Fri, 19 Jun 2020 07:10:24 +0300
-Subject: [PATCH] stress-ng.h: suppress kernel sysinfo.h
-
-The kernel sysinfo.h (indirectly included from genetlink.h) defines
-struct sysinfo. This collides with musl libc definition of the same
-struct.
-
-Fixes this build issue:
-
-In file included from .../arm-buildroot-linux-musleabihf/sysroot/usr/include/linux/kernel.h:5,
-                 from .../arm-buildroot-linux-musleabihf/sysroot/usr/include/linux/netlink.h:5,
-                 from .../arm-buildroot-linux-musleabihf/sysroot/usr/include/linux/genetlink.h:6,
-                 from stress-ng.h:464,
-                 from stress-access.c:25:
-.../arm-buildroot-linux-musleabihf/sysroot/usr/include/linux/sysinfo.h:8:8: error: redefinition of ‘struct sysinfo’
- struct sysinfo {
-        ^~~~~~~
-In file included from stress-ng.h:389,
-                 from stress-access.c:25:
-.../arm-buildroot-linux-musleabihf/sysroot/usr/include/sys/sysinfo.h:10:8: note: originally defined here
- struct sysinfo {
-        ^~~~~~~
-
-Signed-off-by: Baruch Siach <baruch@tkos.co.il>
----
-Upstream status: https://github.com/ColinIanKing/stress-ng/pull/69
-
- stress-ng.h | 2 ++
- 1 file changed, 2 insertions(+)
-
-diff --git a/stress-ng.h b/stress-ng.h
-index 1a662938c1d9..e8aa0efe7527 100644
---- a/stress-ng.h
-+++ b/stress-ng.h
-@@ -387,6 +387,8 @@
- 
- #if defined(HAVE_SYS_SYSINFO_H)
- #include <sys/sysinfo.h>
-+/* Suppress kernel sysinfo to avoid collision with musl */
-+#define _LINUX_SYSINFO_H
- #endif
- 
- #if defined(HAVE_SYS_SYSMACROS_H)
--- 
-2.27.0
-
diff --git a/package/stress-ng/stress-ng.hash b/package/stress-ng/stress-ng.hash
index 9993c759ba..5250a22475 100644
--- a/package/stress-ng/stress-ng.hash
+++ b/package/stress-ng/stress-ng.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  b21436fdbd9dc482a3fd95ae27cccf0097d0f226361ea3785215f7a4ad50136b  stress-ng-0.11.14.tar.xz
+sha256  860291dd3a18b985b3483190a627bbede2b5c52113766c1921001b3fb4b83af0  stress-ng-0.11.17.tar.xz
 sha256  8177f97513213526df2cf6184d8ff986c675afb514d4e68a404010521b880643  COPYING
diff --git a/package/stress-ng/stress-ng.mk b/package/stress-ng/stress-ng.mk
index 61bc07d60a..57464e9bbf 100644
--- a/package/stress-ng/stress-ng.mk
+++ b/package/stress-ng/stress-ng.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-STRESS_NG_VERSION = 0.11.14
+STRESS_NG_VERSION = 0.11.17
 STRESS_NG_SOURCE = stress-ng-$(STRESS_NG_VERSION).tar.xz
 STRESS_NG_SITE = http://kernel.ubuntu.com/~cking/tarballs/stress-ng
 STRESS_NG_LICENSE = GPL-2.0+
-- 
2.28.0

